storage_controller/src/tenant_shard.rs

```diff
@@ -1844,6 +1850,8 @@ impl TenantShard {
             config: serde_json::from_str(&tsp.config).unwrap(),
             reconciler: None,
             splitting: tsp.splitting,
+            // Filled in during [`Service::startup_reconcile`]
+            importing: TimelineImportState::Idle,
```

Comment:
`:(` not a fan of incorrect default values that get filled in later

Reply:
I think given the few places that do check for this, let's add a third state `::Unset` or `::FilledByStartupReconcile`.

Reply:
Well, `Idle` is actually the correct state for anything that doesn't have an import on-going.
We only override it for shards with an importing timeline. Perhaps the comment can be tweaked.

And yeah, I'm not a fan of this either, but didn't like the extra complexity required just to make myself feel good.